common/lib/xmodule/xmodule/modulestore/split_mongo/split.py

```diff
@@ -933,6 +934,35 @@ def get_courses(self, branch, **kwargs):
         # get the blocks for each course index (s/b the root)
         return self._get_structures_for_branch_and_locator(branch, self._create_course_locator, **kwargs)
 
+    @autoretry_read()
+    def get_courses_summary(self, branch, **kwargs):
+        """
+        Returns a list of `CourseSummaryDescriptor` which matching any given qualifiers.
+
+        qualifiers should be a dict of keywords matching the db fields or any
+        legal query for mongo to use against the active_versions collection.
+
+        Note, this is to find the current head of the named branch type.
+        To get specific versions via guid use get_course.
+
+        :param branch: the branch for which to return courses.
+        """
+        courses_summaries = []
+        for entry, structure_info in self._get_structures_for_branch(branch, **kwargs):
+            course_locator = self._create_course_locator(structure_info, branch=None)
+            course_block = [
+                block_data
+                for block_key, block_data in entry['blocks'].items()
+                if block_key.type == "course"
+            ]
+            if not course_block:
+                raise ItemNotFoundError
```

Comment:
@adampalay What exception we should raise if in case we find multiple `course` blocks?